4.8.0 faills to build with networkmanager=true, modemmanager=false

```
 * cinnamon-control-center version 4.8.0
 * Distribution - Gentoo Linux (binaryden overlay)
 * Graphics hardware *and* driver used: NVIDIA GTX 1050 Ti, nvidia-drivers 455.38
 * 64 bit
 ```

**Issue**
Compilation fails for the specific combination of meson options `networkmanager=true` and `modemmanager=false`:
```
[96/99] Linking target panels/network/libnetwork.so
FAILED: panels/network/libnetwork.so 
cc  -o panels/network/libnetwork.so panels/network/libnetwork.so.p/meson-generated_.._cc-network-resources.c.o panels/network/libnetwork.so.p/cc-network-panel.c.o panels/network/libnetwork.so.p/net-device-ethernet.c.o panels/network/libnetwork.so.p/net-device-simple.c.o panels/network/libnetwork.so.p/net-device-wifi.c.o panels/network/libnetwork.so.p/net-device.c.o panels/network/libnetwork.so.p/net-object.c.o panels/network/libnetwork.so.p/net-proxy.c.o panels/network/libnetwork.so.p/net-vpn.c.o panels/network/libnetwork.so.p/network-dialogs.c.o panels/network/libnetwork.so.p/network-module.c.o panels/network/libnetwork.so.p/panel-common.c.o -Wl,--as-needed -Wl,--no-undefined -shared -fPIC -Wl,--start-group -Wl,-soname,libnetwork.so -Wl,--whole-archive panels/network/connection-editor/libconnection_editor.a -Wl,--no-whole-archive shell/libcinnamon-control-center.so.1.0.0 panels/network/wireless-security/libwireless_security.a /usr/lib64/libglib-2.0.so /usr/lib64/libgtk-3.so /usr/lib64/libgdk-3.so /usr/lib64/libpangocairo-1.0.so /usr/lib64/libpango-1.0.so /usr/lib64/libatk-1.0.so /usr/lib64/libcairo-gobject.so /usr/lib64/libcairo.so /usr/lib64/libgdk_pixbuf-2.0.so /usr/lib64/libgio-2.0.so /usr/lib64/libgobject-2.0.so /usr/lib64/libnm.so /usr/lib64/libnma.so /usr/lib64/libpolkit-gobject-1.so -Wl,--end-group '-Wl,-rpath,$ORIGIN/../../shell' -Wl,-rpath-link,/home/thomas/tmp/cinnamon-control-center/build/shell
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: panels/network/libnetwork.so.p/cc-network-panel.c.o: in function `panel_add_device':
/home/thomas/tmp/cinnamon-control-center/build/../panels/network/cc-network-panel.c:655: undefined reference to `net_device_mobile_get_type'
collect2: error: ld returned 1 exit status
```

**Steps to reproduce**
`meson <src_folder> -Dnetworkmanager=true -Dmodemmanager=false && ninja`


**Expected behaviour**
Build success.


**Other information**
The function in question (`net_device_mobile_get_type`) is provided by `net-devices-mobile.c`, which is only included when modemmanager is found. 
https://github.com/linuxmint/cinnamon-control-center/blob/13477da648f4f0771ebe4b397f9f825c0868014c/panels/network/meson.build#L23-L25

However, the function is used unconditionally in 
https://github.com/linuxmint/cinnamon-control-center/blob/13477da648f4f0771ebe4b397f9f825c0868014c/panels/network/cc-network-panel.c#L655
leading to the error above.

--- a/panels/network/cc-network-panel.c
+++ b/panels/network/cc-network-panel.c
@@ -29,7 +29,9 @@
 #include <NetworkManager.h>
 
 #include "net-device.h"
+#ifdef BUILD_MODEM
 #include "net-device-mobile.h"
+#endif
 #include "net-device-wifi.h"
 #include "net-device-ethernet.h"
 #include "net-object.h"
@@ -651,9 +653,11 @@ panel_add_device (CcNetworkPanel *panel, NMDevice *device)
         case NM_DEVICE_TYPE_ETHERNET:
                 device_g_type = NET_TYPE_DEVICE_ETHERNET;
                 break;
+        #ifdef BUILD_MODEM
         case NM_DEVICE_TYPE_MODEM:
                 device_g_type = NET_TYPE_DEVICE_MOBILE;
                 break;
+        #endif
         case NM_DEVICE_TYPE_WIFI:
                 device_g_type = NET_TYPE_DEVICE_WIFI;
                 break;

